darts/models/regression_model.py

```diff
@@ -84,11 +84,41 @@ def predict(self, features: List[TimeSeries]) -> TimeSeries:
                      'The number and dimensionalities of all given features must correspond to those used for'
                      ' training.', logger)
 
+    def _backtest_sanity_checks(self, *args: Any, **kwargs: Any) -> None:
+        """Sanity checks for the backtest function
+
+        Parameters
+        ----------
+        args
+            The args parameter(s) provided to the backtest function.
+        kwargs
+            The kwargs paramter(s) provided to the backtest function.
+
+        Raises
+        ------
+        ValueError
+            when a check on the parameter does not pass.
+        """
+
+        # parse args
+        feature_series = args[0]
+        target_series = args[1]
+
+        raise_if_not(all([s.has_same_time_as(target_series) for s in feature_series]), 'All provided time series must '
+                     'have the same time index', logger)
+
+        _backtest_general_checks(target_series, kwargs)
+
+    def _backtest_model_specific_sanity_checks(self, *args: Any, **kwargs: Any) -> None:
+        """Method to be overriden in subclass for model specific sanity checks"""
+        pass
+
+    @_with_sanity_checks("_backtest_sanity_checks", "_backtest_model_specific_sanity_checks")
     def backtest(self,
                  feature_series: Iterable[TimeSeries],
                  target_series: TimeSeries,
-                 start: pd.Timestamp,
-                 forecast_horizon: int,
+                 start: Union[pd.Timestamp, float, int] = 1,
```

Comment:
Not too sure about having a default value for this parameter. Or if we have one, we should be consistent with ForecastingModel and put 0.5. (although I think 0.7 on both models would be better, actually)

Reply:
I wasn't too sure either.
The problem I see with making it 0.5 (or 0.7) as in ForecastingModel is that in the both model don't "work" with the same input series:

- In the ForecastingModel case, the series `backtest()` is "working" on (ie. `training_series`) is typically the "whole" series dataset (e.g. from 2000-01-01 to 2020-01-01)
- Whereas RegressionModel uses  `features_series` which are mostly already forecasts of other models (and would e.g. go each go from **2010**-01-01 to 2020-01-01.

I think the typical use case of RegressionModel would thus have a `start` param pointing to the very beginning of `feature_series`, and not at 50% or 70%

Reply:
Fair point, however there is still the notion of training and validation set. So I think it remains intuitive (at least to me) that you would train regression models on the first 70% of the data available (so from 2010 to 2017) and then validate on the recent past only. Training on 1 data point is a bit awkward. So I'd vote for putting 70% everywhere or, if you really don't like it, to drop this default value altogether from `RegressionModel.backtest()`.

Reply:
Fair enough, I think we should go for 70% by default everywhere then.